Pre-emptively check SDK version at start

## Background

One failure case is where a user runs `make start` but does not have the appropriate SDK installed. For example, after upgrade of Splice that requires a new SDK version:
```
> Task :daml:compileDaml
Running multi-package build of all packages in /Users/curtishrischuk/src/cn-quickstart-os/quickstart/daml.
damlc: option -W: Warning flag is not valid - warning flags must be of the form `-Werror=<name>`, `-Wno-<name>`, or `-W<name>`. Available names are: upgrade-interfaces
Usage: daml build [--include INCLUDE-PATH] [--package-db LOC-OF-PACKAGE-DB]
                  [--access-token-file PATH]
                  [--shake-profiling PROFILING-REPORT] [[--jobs THREADS]]
                  [--debug | [--log-level ARG]] [--detail LEVEL]
                  [--ghc-option OPTION] [(-p|--test-pattern PATTERN)]
                  [--typecheck-upgrades ARG] [--upgrades UPGRADE_DAR] [-W ARG]
                  [-o|--output FILE] [--incremental ARG] [--init-package-db ARG]
CallStack (from HasCallStack):
  error, called at compiler/damlc/lib/DA/Cli/Damlc.hs:1721:20 in compilerZSdamlcZSdamlc-lib:DA.Cli.Damlc
> Task :daml:compileDaml FAILED
FAILURE: Build failed with an exception.
```

Resolution: `make install-daml-sdk`

We can do a pre-emptive check that the SDK is installed.

## Task

Create a new make target `make check`, and have `make start` include it. This dependency check should look at the installed SDK version (`daml version`) and ensure the [SDK version](https://github.com/digital-asset/cn-quickstart/blob/main/quickstart/.env#L5) is present. Instruct the user on the version mismatch (display both what is installed, and what is required) - and direct them to run `make install-daml-sdk`.

--- a/quickstart/Makefile
+++ b/quickstart/Makefile
@@ -241,6 +241,10 @@ clean-all: clean clean-docker ## Stop and remove all build artifacts, Docker con
 install-daml-sdk: ## Install the Daml SDK
 	./gradlew :daml:installDamlSdk
 
+.PHONY: check-daml-sdk
+check-daml-sdk: ## Install the Daml SDK
+	./gradlew :daml:verifyDamlSdkVersion
+
 .PHONY: fetch-grafana-dashboards
 fetch-grafana-dashboards: ## Download Grafana dashboards
 	./gradlew fetchGrafanaDashboards

--- a/quickstart/daml/build.gradle.kts
+++ b/quickstart/daml/build.gradle.kts
@@ -18,6 +18,7 @@ plugins {
 }
 
 tasks.register<Exec>("compileDaml") {
+    dependsOn("verifyDamlSdkVersion")
     commandLine("daml", "damlc", "build", "--all")
 }
 
@@ -131,5 +132,52 @@ tasks.register<Exec>("installDamlSdk") {
         println("Installed Daml SDK runtime $damlSdkRuntimeVersion as $damlSdkVersion")
         println("Cleaning up downloaded files")
         sdkDir.deleteRecursively()
+    }
+}
+
+// Task to ensure the right Daml SDK version is installed
+tasks.register("verifyDamlSdkVersion") {
+    val sdkVars = computeSdkVariables()
+    val requiredVersion = sdkVars["damlSdkVersion"] as String
+
+    doLast {
+        val output = java.io.ByteArrayOutputStream()
+        exec {
+            commandLine = listOf("daml", "version")
+            standardOutput = output
+            isIgnoreExitValue = true
+        }
+
+        val versionLine = output.toString()
+            .lineSequence()
+            .firstOrNull { it.contains("default SDK version") }
+            ?.trim()
+
+        val installedVersion = versionLine?.replace(Regex("\\s*\\(default SDK version.*\\)\\s*"), "")?.trim()
+
+        if (installedVersion == null) {
+            throw GradleException(
+                """
+                ❌ Could not determine DAML SDK version. 
+                   Is it installed and on your PATH?
+
+                   💡 Please try running: make install-daml-sdk
+                   """.trimIndent())
+        }
+
+        if (installedVersion != requiredVersion) {
+            throw GradleException(
+                """
+                ❌ DAML SDK version mismatch:
+                   Required:  $requiredVersion
+                   Installed: $installedVersion
+                
+                💡 Please try running: make install-daml-sdk
+                """.trimIndent()
+            )
+        } else {
+            println("✅ DAML SDK version $installedVersion is installed.")
+        }
+
     }
 }

--- a/quickstart/daml/build.gradle.kts
+++ b/quickstart/daml/build.gradle.kts
@@ -161,8 +161,8 @@ tasks.register("verifyDamlSdkVersion") {
                 ❌ Could not determine DAML SDK version. 
                    Is it installed and on your PATH?
 
-                   💡 Please try running: make install-daml-sdk
-                   """.trimIndent())
+                💡 Please try running: make install-daml-sdk
+                """.trimIndent())
         }
 
         if (installedVersion != requiredVersion) {

